design/httpproxy-header-rewrite.md

```diff
@@ -0,0 +1,79 @@
+# Header Rewrite For HTTPProxy
+
+**Status**: _Draft_
+
+This document specifies a design for supporting request / response header manipulation in the HTTPProxy CRD.
+
+## Goals
+
+Setting and removing headers from requests and responses at two levels:
+1. Per-route (pre-split),
+1. Per-cluster (post-split).
+
+## Non Goals
+
+- Support for projecting dynamic values into header values.
+- Appending to pre-existing headers (for now).
+
+## Background
+
+There are a number of use cases where having this raw capability is useful, and many are called out in #70.
+
+## High-Level Design
+
+Add a new type: `HeaderPolicy` that captures setting and removing headers.
+
+```Go
+// HeaderPolicy defines alterations to the headers being sent to or returned from a service.
+type HeaderPolicy struct {
+	// Set sets the specified headers replacing any existing values associated with the header names.
+	// +optional
+	Set []HeaderAddition `json:"set,omitempty"`
+	// Remove removes any headers whose name matches those specified.
+	// +optional
+	Remove []string `json:"remove,omitempty"`
+}
+
+// HeaderAddition defines a header key/value pair to be added to those sent to or return from a service.
+type HeaderAddition struct {
```

Comment:
Agreed with a comment on Slack that this should be called something like 'HeaderValue' since it's a generic representation of a full header.

Reply:
done